include/logger.hrl

```diff
@@ -18,33 +18,55 @@
 
 -compile({parse_transform, emqx_logger}).
```

Comment:
Why not deprecate the entire header and use kernel logger instead?

Reply:
Could you give an example?

Reply:
What I mean, why not try to migrate to `?LOG(` macros from `kernel/include/logger.hrl` to reduce the amount of custom code? I think the only distinguishing feature of emqx logger header is custom prefix. I thing it could be emulated by tweaking logger handler.

Reply:
not sure what's easier. 
making a formatter, or use `report_cb`

anyway, will address this in another PR.